src/pages/[platform]/build-a-backend/storage/storage-v5-to-v6-migration-guide/index.mdx

```diff
@@ -0,0 +1,1242 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Migrate from v5 to v6',
+  description: 'Learn more about the migration steps to upgrade Storage APIs for Amplify JavaScript v5 to v6',
+  platforms: ['angular', 'javascript', 'nextjs', 'react', 'vue', 'react-native']
+};
+
+export const getStaticPaths = async () => {
+  return getCustomStaticPath(meta.platforms);
+};
+
+export function getStaticProps(context) {
+  return {
+    props: {
+      platform: context.params.platform,
+      meta
+    }
+  };
+}
+
+This guide will help you migrate v5 Storage implementations to the new functional API's in v6.
+
+<Callout>Many input and output properties have been deprecated due to low usage, added complexity, or being deemed non-critical. If you would like to see these options re-implemented in v6, feel free to open a [feature request](https://github.com/aws-amplify/amplify-js/issues/new?assignees=&labels=pending-triage&projects=&template=2.feature_request.yaml) in our github repository. Keep in mind that an issue may already be open and that commenting on an existing feature request increases the likelihood that it will be implemented in a future release.</Callout>
+
+The input options below have been removed in v6 from Storage API's:
+
+- `provider`: _Custom providers are no longer supported in v6._
+- `customPrefix`: _You can [add a prefix resolver](/[platform]/build-a-backend/storage/configure-access/#customize-object-key-path) to your Storage Config to customize the key prefix per accessLevel. See the [Using a Custom Prefix](/[platform]/build-a-backend/storage/storage-v5-to-v6-migration-guide/#using-a-custom-prefix) header for migration details._
+- `track`: _Tracking Storage events in Hub is not yet supported in v6._
+- `ServerSideEncryption`, `SSECustomerAlgorithm`, `SSECustomerKey`, `SSECustomerKeyMD5`, and `SSEKMSKeyId`: _These options have been deprecated due to low usage and added complexity: Server-Side Encryption has been automatically applied using default keys by S3 [since Jan 2023](https://aws.amazon.com/blogs/aws/amazon-s3-encrypts-new-objects-by-default/)._
+- `cacheControl`: _Response payload is always set to 'no-store' in v6._
+- `$metadata`: _Information like `statusCode` and `retryCount` will be provided in v6 via debug messages using [the Console Logger](/[platform]/build-a-backend/utilities/console-logger/)._
+
+The `progressCallback` option on upload and download operations has been replaced by `onProgress` in v6. Note that the parameter property names have also changed.
+
+## Storage.put
+
+The `Storage.put` API has moved to the `uploadData` API in v6. There are a few changes in the experience from v5 to v6 in addition to those mentioned above:
+
+- We have remapped all capitalized input and output keys to camel case.
+- The `cacheControl` option has been removed and the response payload is always set to `no-store` in v6.
```

Comment:
This was changed in later revision([link](https://quip-amazon.com/acftAfZE4PGI/Amplify-JS-v6-Storage-API-Rev3#temp:C:LZE3569bab6cb144f50b886f2a65)). 

```suggestion
- The `cacheControl` option has been removed and now defers to the HTTP caching behavior configured by the runtime.
```